q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,304 @@
+
+    Attributes
+    ----------
+    rules: dict
+        Checked rules in a hard-coded data structure that
+        has common default values for all variables and
+        that is then updated based on the parsed rules.
+    variables_rules : dict
+        Rules (values) parsed from the current variable's yml file.
+    warnings : WarningsCollection instance
+        Collection of warnings encountered during either the parsing
+        of the variables rules, or during the application the rules.
+    errors : ErrorsCollection instance
+        Collection of errors encountered during either the parsing
+        of the variables rules, or during the application the rules.
+
+    Methods
+    -------
+    parse_rule : staticmethod
+        Performs the parsing of the current variable's yml file and
+        returns the rules (values) in `self.variables_rules`.
+    _check_variable_rules
+        Loops over each rule and check that they are well formatted
+        and either raise a specific issue error or a fill the rules dict.
+    normalize
+        Perform the application of the variables' rules.
+
+    Raises
+    ------
+    IOError : yaml.YAMLError
+        If something went wrong with reading the .yml rule file.
+    ExpectedError
+        If wrong formatting of an "expected" rule.
+    OntologyError
+        If wrong formatting of an "ontology" rule.
+    RemapError
+        If wrong formatting of a "remap" rule.
+    ValidationError
+        If wrong formatting of a "validation" rule.
+    NormalizationError,
+         If wrong formatting of an "normalization" rule.
+    BlankError
+        If wrong formatting of a "blank" rule.
+    MissingError
+        If wrong formatting of a "missing" rule.
+    FormatError
+        If wrong formatting of a "format" rule.
+    """
+    def __init__(self, variable_rules_fp: str):
+        """Initialize the class instance for the set of rules associated
+        with a single metadata variable. The instantiated class will be
+        given a `variable` attribute corresponding to the variable name
+        as string, as well as a `rules` attribute collecting the actual
+        rules in a dictionary.
+
+        Parameters
+        ----------
+        variable_rules_fp : str
+            Path to one variable's rules yaml file.
+        """
+        self.rules = {
+            'edits': {
+                'remap': {},
+                'normalization': {},
+                'validation': {}
+            },
+            'lookups': {
+                'ontology': None,
+                'expected': []
+            },
+            'allowed': {
+                'blank': None,
+                'missing': None,
+                'format': None
+            }
+
+        }
+        self.parsed_rules = self.parse_rule(variable_rules_fp)
+        self.warnings = WarningsCollection()
+        self.errors = ErrorsCollection()
+
+    @staticmethod
+    def parse_rule(variable_rules_fp: str):
```

Comment:
it looks like this parses yaml but doesn't care what the yaml contains

Reply:
yes, because the check happens after.
See https://github.com/ElDeveloper/q2-metadata/pull/3#discussion_r414976404 please